bezier-rs/lib/src/lib.rs

```diff
@@ -305,4 +305,58 @@ impl Bezier {
 		};
 		bezier_starting_at_t1.split(adjusted_t2)[t2_split_side]
 	}
+
+	/// Returns two lists of t-values representing the local extrema of the x and y parametric curves respectively
+	/// The local extrema are defined to be points at which the derivative of the curve is equal to zero
+	fn _local_extrema(&self) -> [Vec<f64>; 2] {
+		let mut x_values = Vec::new();
+		let mut y_values = Vec::new();
+		match self.handles {
+			BezierHandles::Quadratic { handle } => {
+				let a = handle - self.start;
+				let b = self.end - handle;
+				// check for x extrema
+				if (b.x - a.x) != 0. {
+					x_values.push(-a.x / (b.x - a.x));
+				}
+				// check for y extrema
+				if (b.y - a.y) != 0. {
+					y_values.push(-a.y / (b.y - a.y));
+				}
+			}
+			BezierHandles::Cubic { handle_start, handle_end } => {
+				let a = 3. * (-self.start + 3. * handle_start - 3. * handle_end + self.end);
+				let b = 6. * (self.start - 2. * handle_start + handle_end);
+				let c = 3. * (handle_start - self.start);
+				// check for x extrema
+				let x_discr = b.x * b.x - 4. * a.x * c.x;
```

Comment:
Minor comment: Again, you could compute the discriminants in a `DVec2` and the use the `.x` and `.y` results for comparisons after.

Also use `discriminant` instead of `discr` for clarity.